File: core/src/test/java/com/larryhsiao/nyx/jots/JotUriIdTest.java
package com.larryhsiao.nyx.jots;

import com.larryhsiao.nyx.core.jots.JotUriId;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Unit-test for the class {@link JotUriId}
 */
public class JotUriIdTest {

    /**
     * Check the output
     */
    @Test
    public void simple() {
        Assertions.assertEquals(
            1L,
            new JotUriId(
                "http://localhost.com/jots/1"
            ).value().longValue()
        );
    }

    /**
     * Not Jot path
     */
    @Test
    public void notJotPath() {
        try {
            new JotUriId("http://localhost.com/abc/1").value();
            Assertions.fail("Should throw exception");
        } catch (Exception e) {
            Assertions.assertTrue(true);
        }
    }

    /**
     * URI id not a long
     */
    @Test
    public void notId() {
        try {
            new JotUriId("http://localhost.com/jots/number").value();
            Assertions.fail("Should throw exception");
        } catch (Exception e) {
            Assertions.assertTrue(true);
        }
    }
}

File: core/src/test/java/com/larryhsiao/nyx/core/jots/JotUriIdTest.kt
package com.larryhsiao.nyx.core.jots

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Unit-test for the class [JotUriId]
 */
class JotUriIdTest {
    /**
     * Check the output
     */
    @Test
    fun simple() {
        Assertions.assertEquals(
            1L,
            JotUriId(
                "http://localhost.com/jots/1"
            ).value()
        )
    }

    /**
     * Not Jot path
     */
    @Test
    fun notJotPath() {
        try {
            JotUriId("http://localhost.com/abc/1").value()
            Assertions.fail<Any>("Should throw exception")
        } catch (e: Exception) {
            Assertions.assertTrue(true)
        }
    }

    /**
     * URI id not a long
     */
    @Test
    fun notId() {
        try {
            JotUriId("http://localhost.com/jots/number").value()
            Assertions.fail<Any>("Should throw exception")
        } catch (e: Exception) {
            Assertions.assertTrue(true)
        }
    }
}